src/OpenTelemetry.Exporter.Prometheus.Shared/PrometheusExporter.cs

```diff
@@ -69,23 +61,25 @@ public PrometheusExporter(PrometheusExporterOptions options)
             set => this.funcExport = value;
         }
 
+        internal Action OnDispose { get; set; }
+
         internal PrometheusCollectionManager CollectionManager { get; }
 
+        /// <inheritdoc/>
         public override ExportResult Export(in Batch<Metric> metrics)
         {
             return this.OnExport(metrics);
         }
 
+        /// <inheritdoc/>
         protected override void Dispose(bool disposing)
         {
             if (!this.disposed)
             {
                 if (disposing)
                 {
-                    this.metricsHttpServer?.Dispose();
+                    this.OnDispose?.Invoke();
                 }
-
-                this.disposed = true;
```

Comment:
Set `this.disposed` to `true`?

Reply:
holy.... don't tell anyone :sweat_smile:



<!--EndFragment-->
</body>
</html>